addons/account/tests/test_early_payment_discount.py

```diff
@@ -728,3 +728,59 @@ def test_mixed_epd_with_tax_deleted_line(self):
         inv.invoice_line_ids[1].unlink()
         self.assertEqual(len(inv.line_ids), 6) # 1 prod, 1 tax, 2 epd, 1 epd tax discount, 1 payment terms
         self.assertEqual(inv.amount_tax, 9.00) # $100.0 @ 10% tax (-10% epd)
+
+    def test_mixed_epd_with_rounding_issue(self):
+        """
+        Ensure epd line will not unbalance the invoice
+        """
+        self.env.company.early_pay_discount_computation = 'mixed'
+        tax_6 = self.env['account.tax'].create({
+             'name': '6%',
+             'amount': 6,
+        })
+        tax_12 = self.env['account.tax'].create({
+             'name': '12%',
+             'amount': 12,
+        })
+        tax_136 = self.env['account.tax'].create({
+             'name': '136',
+             'amount': 0.136,
+             'amount_type': 'fixed',
+             'include_base_amount': True,
+        })
+        tax_176 = self.env['account.tax'].create({
+             'name': '176',
+             'amount': 0.176,
+             'amount_type': 'fixed',
+             'include_base_amount': True,
+        })
+
+        early_pay_1_percents_7_days = self.env['account.payment.term'].create({
+            'name': '1% discount if paid within 7 days',
+            'company_id': self.company_data['company'].id,
+            'line_ids': [Command.create({
+                'value': 'balance',
+                'days': 0,
+                'discount_percentage': 1,
+                'discount_days': 7,
+            })]
+        })
+
+        with Form(self.env['account.move'].with_context(default_move_type='out_invoice')) as invoice_form:
+            invoice_form.partner_id = self.partner_a
+            invoice_form.invoice_date = fields.Date.from_string('2022-02-21')
+            invoice_form.invoice_payment_term_id = early_pay_1_percents_7_days
+            for price_unit, quantity, taxes in [
+                (116, 6, tax_6),
+                (0.91, 350, tax_6),
+                (194.21, 1, tax_136 | tax_12),
+                (31.46, 5, tax_176 | tax_12)
+            ]:
+                with invoice_form.invoice_line_ids.new() as line_form:
+                    line_form.price_unit = price_unit
+                    line_form.quantity = quantity
+                    for tax in taxes:
+                        line_form.tax_ids.add(tax)
+
+        # If invoice is not balanced the following line will raise
+        invoice_form.save()
```

Comment:
the save is already triggered when exiting the context manager so this is useless fyi